File: service/src/main/java/com/coopra/service/service_implementations/TracksService.java
package com.coopra.service.service_implementations;

import com.coopra.data.Track;
import com.coopra.service.Constants;
import com.coopra.service.RetrofitHelper;
import com.coopra.service.interfaces.TracksInterface;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;

public class TracksService {
    public static void getRandomTracks(Callback<List<Track>> callback) {
        TracksInterface service = RetrofitHelper.createRetrofit().create(TracksInterface.class);

        Call<List<Track>> call = service.getRandomTracks(Constants.CLIENT_ID);
        call.enqueue(callback);
    }
}

File: service/src/main/java/com/coopra/service/service_implementations/TracksService.kt
package com.coopra.service.service_implementations

import com.coopra.data.Origin
import com.coopra.service.BuildConfig
import com.coopra.service.RetrofitHelper
import com.coopra.service.interfaces.TracksInterface
import retrofit2.Callback

class TracksService {
    private val retrofitHelper = RetrofitHelper()

    fun getRandomTracks(callback: Callback<List<Origin>>) {
        val service = retrofitHelper.createRetrofit().create(TracksInterface::class.java)
        val call = service.getRandomTracks(BuildConfig.CLIENT_ID)
        call.enqueue(callback)
    }
}